drivers/sensor/bosch/bmp180/bmp180.c

```diff
@@ -0,0 +1,463 @@
+
+	/* tmp is Pa. Convert to kPa as specified in sensor
+	 * interface.
+	 */
+	val->val1 = tmp / 1000;
+	val->val2 = (tmp % 1000) * 1000;
+
+	return 0;
+}
+
+static int bmp180_channel_get(const struct device *dev,
+			      enum sensor_channel chan,
+			      struct sensor_value *val)
+{
+	switch (chan) {
+	case SENSOR_CHAN_PRESS:
+		bmp180_press_channel_get(dev, val);
+		break;
+
+	case SENSOR_CHAN_DIE_TEMP:
+		bmp180_temp_channel_get(dev, val);
+		break;
+
+	default:
+		LOG_DBG("Channel not supported.");
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+
+static int bmp180_get_calibration_data(const struct device *dev)
+{
+	struct bmp180_data *data = dev->data;
+	struct bmp180_cal_data *cal = &data->cal;
+
+	if (bmp180_reg_read(dev, BMP180_REG_CALIB0, (uint8_t *)cal,
+			    sizeof(struct bmp180_cal_data)) < 0) {
+		return -EIO;
+	}
+
+	cal->ac1 = BSWAP_s16(cal->ac1);
+	cal->ac2 = BSWAP_s16(cal->ac2);
+	cal->ac3 = BSWAP_s16(cal->ac3);
+	cal->ac4 = BSWAP_u16(cal->ac4);
+	cal->ac5 = BSWAP_u16(cal->ac5);
+	cal->ac6 = BSWAP_u16(cal->ac6);
+	cal->b1 = BSWAP_s16(cal->b1);
+	cal->b2 = BSWAP_s16(cal->b2);
+	cal->mb = BSWAP_s16(cal->mb);
+	cal->mc = BSWAP_s16(cal->mc);
+	cal->md = BSWAP_s16(cal->md);
+
+	return 0;
+}
+
+#ifdef CONFIG_PM_DEVICE
+static int bmp180_pm_action(const struct device *dev,
+			    enum pm_device_action action)
+{
+	/* no power saving feature in bmp180 */
+	return 0;
+}
+#endif /* CONFIG_PM_DEVICE */
+
+static const struct sensor_driver_api bmp180_api = {
+	.attr_set = bmp180_attr_set,
+	.sample_fetch = bmp180_sample_fetch,
+	.channel_get = bmp180_channel_get,
+};
+
+static int bmp180_init(const struct device *dev)
+{
+	uint8_t val = 0U;
+
+	if (bmp180_bus_check(dev) < 0) {
+		LOG_DBG("bus check failed");
+		return -ENODEV;
+	}
+
+	/* reboot the chip */
+	if (bmp180_reg_write(dev, BMP180_REG_CMD, BMP180_CMD_SOFT_RESET) < 0) {
+		LOG_ERR("Cannot reboot chip.");
+		return -EIO;
+	}
+
+	k_sleep(K_MSEC(2));
+
+	if (bmp180_reg_read(dev, BMP180_REG_CHIPID, &val, 1) < 0) {
+		LOG_ERR("Failed to read chip id.");
+		return -EIO;
+	}
+
+	if (val != BMP180_CHIP_ID) {
+		LOG_ERR("Unsupported chip detected (0x%x)!", val);
+		return -ENODEV;
+	}
+
+	/* Read calibration data */
+	if (bmp180_get_calibration_data(dev) < 0) {
+		LOG_ERR("Failed to read calibration data.");
+		return -EIO;
+	}
+
+	return 0;
+}
+
+#define BMP180_INST(inst)						   \
+	static struct bmp180_data bmp180_data_##inst = {		   \
+		.osr_pressure = DT_INST_ENUM_IDX(inst, osr_press),	   \
+	};								   \
+	static const struct bmp180_config bmp180_config_##inst = {	   \
+		.i2c = I2C_DT_SPEC_INST_GET(inst),			   \
+	};								   \
+	PM_DEVICE_DT_INST_DEFINE(inst, bmp180_pm_action);		   \
+	SENSOR_DEVICE_DT_INST_DEFINE(					   \
+		inst,							   \
+		bmp180_init,						   \
+		PM_DEVICE_DT_INST_GET(inst),				   \
+		&bmp180_data_##inst,					   \
+		&bmp180_config_##inst,					   \
+		POST_KERNEL,						   \
+		CONFIG_SENSOR_INIT_PRIORITY,				   \
+		&bmp180_api);
```

Comment:
Looks nice, but I don't think it'll pass clang-format. Can you run format on this file?

Reply:
This is same same in other drivers, no problem remove the tabs and spaces. SENSOR_DEVICE_DT_INST_DEFINE is a macro, right. The compiler will remove the spaces anyway.

Reply:
I guess its not about the compiler  @yperessrather rather thinks about style consistency. As of now there is no hard rule afaik if clang-format is required. Its up to the maintainers/collaborators to decide if the formatting is ok. Personally I find that clang-format makes it easy and reduces the amount of discussion around formatting if used.

Reply:
@faxe1008 , @yperess 
then please give me a direction, I did't see any complains in the clang check for this macro(s). If you fine for time being may you consider to resolve this conversation. Thank both of you for your time review the driver!!

Reply:
changed BMP180_INST() to clang format.

Reply:
If you want to keep the format just wrap it with

```
/* clang-format off */

...

/* clang-format on */
```

The reason I ask for clang-format to be run on all files is that many of us (not all) have clang-format run automatically on save. So if I have to go in and edit this file my change will include a bunch of format changes which makes bug fixes messy. As a rule, run clang-format on everything. If you hate the way it looks add the ignore comment blocks around it.